modules/tensor_mechanics/include/materials/ComputeLinearViscoelasticCreepStrain.h

```diff
@@ -0,0 +1,70 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef COMPUTELINEARVISCOELASTICCREEPSTRAIN_H
+#define COMPUTELINEARVISCOELASTICCREEPSTRAIN_H
+
+#include "ComputeCreepStrainBase.h"
+#include "LinearViscoelasticityBase.h"
+
+class ComputeLinearViscoelasticCreepStrain;
+
+template <>
+InputParameters validParams<ComputeLinearViscoelasticCreepStrain>();
+
+/**
+ * Computes the creep strains resulting from a spring-dashpot model.
+ * The spring-dashpot model itself is stored in a LinearViscoelasticityBase object,
+ * which performs the update and computation of the creep strain itself.
+ *
+ * The creep strain is separated into a series of internal viscous strain,
+ * on a one-by-one basis for each dashpot in the spring-dashpot model.
+ *
+ * The creep strain can also be drived by an eigenstrain, in which case
+ * the eigenstrain acts as an additional stress on the spring-dashpot model,
+ * and therefore increases the creep strain accordingly.
+ */
+class ComputeLinearViscoelasticCreepStrain : public ComputeCreepStrainBase
+{
+public:
+  ComputeLinearViscoelasticCreepStrain(const InputParameters & parameters);
+
+  // calls the spring-dashpot model to update the viscous strains
```

Comment:
if you use `///` instead of `//` for documenting the methods in the header the comments get picked up by Doxygen and put in the code documentation system